src/Components/Falling.cs

```diff
@@ -0,0 +1,258 @@
+            }
+
+            // particles
+            self.FallParticles?.Invoke();
+
+            // fall
+            float speed = 0f;
+            float maxSpeed = self.FallSpeed;
+            while (true)
+            {
+                // update the speed
+                speed = Calc.Approach(speed, maxSpeed, 500f * Engine.DeltaTime);
+                // try to move
+                if (!entity.MoveVCollideSolids(speed * Engine.DeltaTime * (FallUp ? -1 : 1), true))
+                {
+                    // if we've fallen out the bottom of the screen, we should remove the entity
+                    // otherwise yield for a frame and loop
+                    if (!FallUp && entity.Top <= level.Bounds.Bottom + 16 && (entity.Top <= level.Bounds.Bottom - 1 || !entity.CollideCheck<Solid>(entity.Position + Vector2.UnitY)) ||
+                        FallUp && entity.Bottom >= level.Bounds.Top - 16 && (entity.Bottom >= level.Bounds.Top + 1 || !entity.CollideCheck<Solid>(entity.Position - Vector2.UnitY)))
+                        yield return null;
+                    else
+                    {
+                        // we've fallen out of the screen and should remove the entity
+                        entity.Collidable = entity.Visible = false;
+                        yield return 0.2f;
+                        if (level.Session.MapData.CanTransitionTo(level, new Vector2(entity.Center.X, FallUp ? (entity.Top - 12f) : (entity.Bottom + 12f))))
+                        {
+                            yield return 0.2f;
+                            level.Shake();
+                            if (ShouldRumble) Input.Rumble(RumbleStrength.Strong, RumbleLength.Medium);
+                        }
+
+                        entity.RemoveSelf();
+                        entity.DestroyStaticMovers();
+                        yield break;
+                    }
+                }
+                else
+                {
+                    // if we hit something, break
+                    break;
+                }
+            }
+
+            // impact effects
+            self.ImpactSfx?.Invoke();
+            if (self.ShouldRumble) Input.Rumble(RumbleStrength.Strong, RumbleLength.Medium);
+            level.DirectionalShake(FallUp ? -Vector2.UnitY : Vector2.UnitY);
+            entity.StartShaking();
+            self.LandParticles?.Invoke();
+            yield return 0.2f;
+            entity.StopShaking();
+
+            // if it's hit the fg tiles then make it safe and end
+            if (EndOnSolidTiles && entity.CollideCheck<SolidTiles>(entity.Position + (FallUp ? -Vector2.UnitY : Vector2.UnitY)))
+            {
+                entity.Safe |= self.ShouldManageSafe;
+                yield break;
+            }
+
+            // wait until we can fall again
+            while (entity.CollideCheck<Platform>(entity.Position + (FallUp ? -Vector2.UnitY : Vector2.UnitY)))
+                yield return 0.1f;
```

Comment:
Is there a reason for this specific wait time? Would `yield return null` work? Or maybe that's how vanilla works.

Reply:
Yeah this is how vanilla does it in `FallingBlock`:
```cs
while (fallingBlock.CollideCheck<Platform>(fallingBlock.Position + new Vector2(0.0f, 1f)))
    yield return (object) 0.1f;
```

Reply:
okay, nevermind then!